src/base/SolutionArray.cpp

```diff
@@ -966,6 +966,143 @@ void SolutionArray::writeHeader(AnyMap& root, const string& id,
+            throw NotImplementedError("SolutionArray::writeEntry",
+                "Detected column name containing double quotes or line feeds: '{}'.",
+                name);
+        }
+        if (name.find(",") != string::npos) {
+            header << "\"" << name << "\"";
+            escaped = true;
+        } else {
+            header << name;
+        }
+        if (col != last) {
+            header << ",";
+        }
+    }
+
+    // (Most) potential exceptions have been thrown; start writing data to file
+    if (std::ifstream(fname).good()) {
+        if (!overwrite) {
+            throw CanteraError("SolutionArray::writeEntry",
+                "File '{}' already exists; use option 'overwrite' to replace CSV file.",
+                fname);
+        }
+        std::remove(fname.c_str());
+    }
+    std::ofstream output(fname);
+    const auto default_precision = output.precision();
+    output << header.str() << std::endl << std::setprecision(9);
+
+    vector<double> buf(speciesNames.size(), 0.);
+    for (size_t row = 0; row < m_size; row++) {
+        setLoc(row);
+        if (mole) {
+            m_sol->thermo()->getMoleFractions(buf.data());
+        } else {
+            m_sol->thermo()->getMassFractions(buf.data());
+        }
+
+        size_t idx = 0;
+        for (size_t col = 0; col < components.size(); col++) {
+            if (isSpecies[col]) {
+                output << buf[idx++];
+            } else {
+                auto& data = components[col];
+                if (data.isVector<double>()) {
+                    output << data.asVector<double>()[row];
+                } else if (data.isVector<long int>()) {
+                    output << data.asVector<long int>()[row];
+                } else {
+                    auto value = data.asVector<string>()[row];
+                    if (value.find("\"") != string::npos ||
+                        value.find("\n") != string::npos)
+                    {
+                        throw NotImplementedError("SolutionArray::writeEntry",
+                            "Detected value containing double quotes or line feeds: "
+                            "'{}'", value);
+                    }
+                    if (value.find(",") != string::npos) {
+                        output << "\"" << value << "\"";
+                        escaped = true;
+                    } else {
+                        output << value;
+                    }
+                }
+            }
+            if (col != last) {
+                output << ",";
+            }
+        }
+        output << std::endl;
+    }
+    output << std::endl << std::setprecision(default_precision);
+
+    if (escaped) {
+        warn_user("SolutionArray::writeEntry",
+            "One or more CSV column names or values contain commas.\n"
+            "Values have been escaped with double quotes, which may not be supported "
+            "by all CSV readers.");
+    }
```

Comment:
I'm not sure this warning will be effective. For one thing, users are unlikely to be able to fix it. For another, they're likely going to forget about it the next day. Unfortunately, there's nothing Cantera can do about ambiguity in the idea of a CSV file. I'd suggest removing this to avoid questions on the UG